### ISSUE TITLE
Missing library artifacts preventing import on other projects

### ISSUE THREAD

**Issue by Project2100** [jitpack/jitpack.io#4110]
Providing the following as requested:

- _"Link to build log"_: [here](https://jitpack.io/com/github/Project2100/java-commons/38efb92da6/build.log)

- _"Does the project build on your machine with **the same** commands (e.g. ./gradlew install)?"_

Actually, my library uses the plugin `maven-publish`, as `maven` is considered deprecated. Therefore, I tried using the `build publishToMavenLocal` variant, which succeeds just as depicted in the log.

Yet...

- _"What error are you seeing?"_

This last line in the log: `ERROR: No build artifacts found`.

Apparently it is important, because I'm trying to use my library in another project that can be found [here](https://github.com/Project2100/Chronos), where I'm referring to the library's latest commit with the `-SNAPSHOT` method (which is Project2100/java-commons@38efb92da6df2c38d74a4e2b1a660ab8b2c12986, as of now).

But, IntelliJ complains about it while synchronizing with the project, with the following message: 

> Could not resolve: com.github.Project2100:java-commons:master-SNAPSHOT
>
>Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
>Use '--warning-mode all' to show the individual deprecation warnings.
>See https://docs.gradle.org/6.0/userguide/command_line_interface.html#sec:command_line_warnings
>
>CONFIGURE SUCCESSFUL in 2s

And then, when issuing a build command for my library-dependent project, this happens:

>18:38:49: Executing tasks ':classes :testClasses'...
>
>> Task :compileJava FAILED
>
>FAILURE: Build failed with an exception.
>
>* What went wrong:
>Execution failed for task ':compileJava'.
>> Could not resolve all files for configuration ':compileClasspath'.
>   > Could not find com.github.Project2100:java-commons:master-SNAPSHOT.
>     Searched in the following locations:
>       - https://jitpack.io/com/github/Project2100/java-commons/master-SNAPSHOT/maven-metadata.xml
>       - https://jitpack.io/com/github/Project2100/java-commons/master-SNAPSHOT/java-commons-master-SNAPSHOT.pom
>     Required by:
>         project :
>
>* Try:
>Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
>
>* Get more help at https://help.gradle.org
>
>BUILD FAILED in 1s
>1 actionable task: 1 executed
>18:38:50: Tasks execution finished ':classes :testClasses'.

I'm not even sure if I'm missing something, as I'm trying to get things to work all day, and I feel stumped at this point. Any help will be much appreciated!

Thank you!

**Comment by jitpack-io** [comment]
Hi,

It seems that the `publishToMavenLocal` is being reported as UP-TO-DATE and doesn't actually publish anything to local maven repository.

According to Gradle docs  https://docs.gradle.org/current/userguide/publishing_maven.html#publishing_maven:publications, 
 the build.gradle.kts needs a publishing section:

```
    publishing {
        publications {
            create<MavenPublication>("maven") {
                from(components["java"])
            }
        }
    }
```

**Comment by Project2100** [comment]
Greetings,

Thank you for your suggestion, I spent some time fixing both repos as I had to update the dependant's references; now I managed to get the latter up and running!

I was convinced that that script snippet was to be inserted exclusively for customization purposes, and wasn't actually required to be there; also I was under the assumption that the 'application' plugin did the publishing job by itself, so no more work was needed by me. My apologies for not being thorough 🙏

Again, many thanks for the help; have a good one!
